TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/EditUser/Deactivate.cshtml.cs

```diff
@@ -0,0 +1,180 @@
+using System.ComponentModel.DataAnnotations;
+using Humanizer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.DataStore.Postgres;
+using TeachingRecordSystem.Core.Services.Files;
+using TeachingRecordSystem.SupportUi.Infrastructure.DataAnnotations;
+using TeachingRecordSystem.SupportUi.Infrastructure.Security;
+using TeachingRecordSystem.SupportUi.Pages.EditUser;
+
+namespace TeachingRecordSystem.SupportUi.Pages.Users.EditUser
+{
+    [Authorize(Policy = AuthorizationPolicies.UserManagement)]
+    public class DeactivateModel(
+        TrsLinkGenerator linkGenerator,
+        IFileService fileService,
+        TrsDbContext dbContext,
+        IClock clock) : PageModel
+    {
+        private Core.DataStore.Postgres.Models.User? _user;
+
+        [FromRoute]
+        public Guid UserId { get; set; }
+
+        [BindProperty]
+        [Display(Name = "Reason for deactivating user")]
+        [Required(ErrorMessage = "Select a reason for deactivating this user")]
+        public bool? HasAdditionalReason { get; set; }
+
+        [BindProperty]
+        [Display(Name = "Enter a reason for deactivating this user")]
+        public string? AdditionalReasonDetail { get; set; }
+
+        [BindProperty]
+        [Display(Name = "Do you have more information?")]
+        [Required(ErrorMessage = "Select yes if you want to provide more details")]
+        public bool? HasMoreInformation { get; set; }
+
+        [BindProperty]
+        [Display(Name = "Enter details")]
+        public string? MoreInformationDetail { get; set; }
+
+        [BindProperty]
+        [Display(Name = "Do you have evidence to upload?")]
+        [Required(ErrorMessage = "Select yes if you want to upload evidence")]
+        public bool? UploadEvidence { get; set; }
+
+        [BindProperty]
+        [EvidenceFile]
+        [FileSize(DeactivateDefaults.MaxFileUploadSizeMb * 1024 * 1024, ErrorMessage = "The selected file must be smaller than 100MB")]
+        public IFormFile? EvidenceFile { get; set; }
+
+        [BindProperty]
+        public Guid? EvidenceFileId { get; set; }
+
+        [BindProperty]
+        public string? EvidenceFileName { get; set; }
+
+        [BindProperty]
+        public string? EvidenceFileSizeDescription { get; set; }
+
+        [BindProperty]
+        public string? UploadedEvidenceFileUrl { get; set; }
+
+        public void OnGet()
+        {
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (_user is null)
```

Comment:
Do we need this check?

Reply:
No but it stops cluttering the code with null handling operators